File: floatinglayout/src/main/java/io/hamed/floatinglayout/FloatingLayout.java
package io.hamed.floatinglayout;

import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import android.os.Looper;

import androidx.annotation.LayoutRes;

import io.hamed.floatinglayout.callback.FloatingListener;
import io.hamed.floatinglayout.service.FloatingService;

/**
 * Author: Hamed Taherpour
 * *
 * Created: 7/30/2020
 * *
 * Address: https://github.com/HamedTaherpour
 */
public class FloatingLayout {

    private Context context;
    private int layoutRes;
    private FloatingListener floatingListener;
    private boolean isShow;
    private Intent intent;

    public FloatingLayout(Context context, @LayoutRes int layoutRes) {
        this.context = context;
        this.layoutRes = layoutRes;
    }

    public void setFloatingListener(FloatingListener floatingListener) {
        this.floatingListener = floatingListener;
    }

    public void create() {
        isShow = true;
        Intent intent = getIntent();
        intent.putExtra(FloatingService.EXTRA_LAYOUT_RESOURCE, layoutRes);
        if (floatingListener != null)
            intent.putExtra(FloatingService.EXTRA_RECEIVER, new FloatingResult(new Handler(Looper.getMainLooper()), floatingListener));
        context.startService(intent);
    }

    public void destroy() {
        isShow = false;
        context.stopService(getIntent());
    }

    public boolean isShow() {
        return isShow;
    }

    private Intent getIntent() {
        if (intent == null)
            intent = new Intent(context, FloatingService.class);
        return intent;
    }
}

File: floatinglayout/src/main/java/io/hamed/floatinglayout/FloatingLayout.kt
package io.hamed.floatinglayout

import android.content.Context
import android.content.Intent
import android.os.Handler
import androidx.annotation.LayoutRes


/**
 * Author: Hamed Taherpour
 * *
 * Created: 12/6/2019
 */
class FloatingLayout(
    mContext: Context,
    @LayoutRes mResource: Int,
    callBack: CallBack
) : FloatingLayoutContract {

    @LayoutRes
    private var mResource: Int = mResource
    private val mContext: Context = mContext
    private var callBack: CallBack = callBack

    private var isShow = false

    override fun onCreate() {
        val intent = Intent(mContext, FloatingLayoutService::class.java)
        intent.putExtra(FloatingLayoutService.LAYOUT_RESOURCE, mResource)
        if (callBack != null)
            intent.putExtra(FloatingLayoutService.RECEIVER, ServiceReceiver(Handler(), callBack))

        mContext.startService(intent)
    }

    override fun onClose() {
        mContext.stopService(Intent(mContext, FloatingLayoutService::class.java))
    }

    override fun create() {
        isShow = true
        onCreate()
    }

    override fun close() {
        isShow = false
        onClose()
    }

    override fun isShow(): Boolean {
        return isShow
    }

}